server/add_dataset.py

```diff
@@ -79,11 +80,12 @@ def getBounds(ds):
 
 def importDataset(dataDir, datafile, description, autoApply=True):
     if not os.path.exists(datafile):
-        print "Data file \"%s\" does not exist" % datafile
+        print("Data file \"%s\" does not exist" % datafile)
         return
     basename = os.path.basename(datafile)
     filedir = os.path.join(dataDir, basename)
-    os.mkdir(filedir)
+    if not os.path.exists(filedir):
+        os.mkdir(filedir)
```

Comment:
I don't recall of the top of my head if there's any chance we could have multiple directories that need to be created.  Should we prefer `os.makedirs` here?  Seems safe since you're checking for existence first.